### ISSUE TITLE
Unable to show dataframe

### ISSUE THREAD

**Issue by vppaidi** [spyder-ide/spyder#21094]
### What steps will reproduce the problem?

Unable to open a data frame in variable explorer. 
Able to open an empty one but unable to open a data frame filled with values

### Traceback
```python-traceback
  File "C:\Users\vpp\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\collectionsdelegate.py", line 199, in createEditor
    if not editor.setup_and_check(value, title=key):
  File "C:\Users\vpp\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\dataframeeditor.py", line 952, in setup_and_check
    self.dataModel = DataFrameModel(data, parent=self)
  File "C:\Users\vpp\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\dataframeeditor.py", line 139, in __init__
    self.max_min_col_update()
  File "C:\Users\vpp\Anaconda3\lib\site-packages\spyder\plugins\variableexplorer\widgets\dataframeeditor.py", line 248, in max_min_col_update
    for dummy, col in self.df.iteritems():
  File "C:\Users\vpp\Anaconda3\lib\site-packages\pandas\core\generic.py", line 5989, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'iteritems'
```

## Versions

* Spyder version: 4.2.5 
* Python version: 3.8.8
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  2.0.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=7.6.0                 :  7.27.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  23.1.0 (OK)
nbconvert >=4.0                 :  6.1.0 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
paramiko >=2.4.0                :  2.7.2 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.8.0 (OK)
pygments >=2.0                  :  2.10.0 (OK)
pylint >=1.0                    :  2.9.6 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.2             :  0.3.2 (OK)
qdarkstyle >=2.8;<3.0           :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.2 (OK)
qtconsole >=5.0.3               :  5.1.1 (OK)
qtpy >=1.5.0                    :  1.10.0 (OK)
rtree >=0.8.3                   :  0.9.7 (OK)
setuptools >=39.0.0             :  67.7.2 (OK)
sphinx >=0.6.6                  :  4.2.0 (OK)
spyder_kernels >=1.10.2;<1.11.0 :  1.10.2 (OK)
textdistance >=4.2.0            :  4.2.1 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0        :  1.0.2 (OK)
zmq >=17                        :  22.2.1 (OK)

# Optional:
cython >=0.21                   :  0.29.33 (OK)
matplotlib >=2.0.0              :  3.4.3 (OK)
numpy >=1.7                     :  1.24.3 (OK)
pandas >=1.1.1                  :  2.0.1 (OK)
scipy >=0.17.0                  :  1.7.1 (OK)
sympy >=0.7.3                   :  1.8 (OK)
```

**Comment by dalthviz** [comment]
Hi @vppaidi thank you for the feedback! Seems like this is a duplicate of https://github.com/spyder-ide/spyder/issues/20496 To get the fix you will need to update to Spyder 5.4.3

Let us know if updating helps!

**Comment by jbnsn** [comment]
Maybe it's a new bug? A colleague of mine and I have the same or a similar problem since a couple of days. I have it on all my Anaconda environments. The version of my Spyder is `Python 3.9.12 64-bit | Qt 5.15.4 | PyQt5 5.15.7 | Windows 10 `.

**Comment by jbnsn** [comment]
I encountered the same issue. How do I install Spyder 5.4.3? I tried `conda install spyder=5.4.3` on a fresh conda environment and got a `PackagesNotFoundError`.

**Comment by dalthviz** [comment]
Hi @jbnsn can you try creating an env with something like `conda create -n spyder-cf -c conda-forge spyder=5.4.3`? Let us know if that enables you to get the latest Spyder version available!

**Comment by jbnsn** [comment]
Thank you. That works fine for me. I was hoping for a solution where I _only_ update Spyder instead of creating a new environment. Anyway, I did it this way now and reinstalled the missing packages. Thank you for your support and a great piece of software!

**Comment by vppaidi** [comment]
> Hi @vppaidi thank you for the feedback! Seems like this is a duplicate of #20496 To get the fix you will need to update to Spyder 5.4.3
> 
> Let us know if updating helps!

**Comment by vppaidi** [comment]
Updating the Spyder resolved this issue. Thanks!
